datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1195,13 +1195,17 @@ pub struct SubqueryAlias {
 }
 
 impl SubqueryAlias {
-    pub fn try_new(plan: LogicalPlan, alias: &str) -> datafusion_common::Result<Self> {
+    pub fn try_new(
+        plan: LogicalPlan,
+        alias: impl Into<String>,
```

Comment:
This change allows `SubqueryAlias::try_new` to take a `String` if the caller has one or a `&str` that will be copied into a new String if needed